src/Microsoft.OData.Client/BaseEntityType.cs

```diff
@@ -9,14 +9,28 @@ namespace Microsoft.OData.Client
     /// <summary>
     /// Base type of entity type to include <see cref="Microsoft.OData.Client.DataServiceContext" /> for function and action invocation
     /// </summary>
-    public class BaseEntityType
+    public class BaseEntityType : IBaseEntityType
     {
-        /// <summary>DataServiceContext for query provider</summary>
+        /// <summary>
+        /// DataServiceContext for query provider.
+        /// </summary>
         protected internal DataServiceContext Context { get; set; }
 
+        DataServiceContext IBaseEntityType.Context
+        {
+            get => Context;
+            set => Context = value;
+        }

```

Comment:
Why make these explicitly implement the interface instead of just making the backing properties public?